### ISSUE TITLE
Add support for JS functions in `plugins` list in Stylelint config

### ISSUE THREAD

**Issue by phoenisx** [stylelint/stylelint#6452]
### What is the problem you're trying to solve?

Using `stylelint` as a library in JS (NodeJS/Deno) does not accept JS functions as `plugins: []` config values. It only works with `string` type for now.

Is it possible to support both?

### What solution would you like to see?

Allow developers to, when using `stylelint` as JS library, to pass locally developed stylelint plugins as config param. For e.g.

```js
stylelint({
  rules: {
    "private/rule-name": true,
  },
  plugins: [require("./plugin-path")],
})
```

I guess this can be resolved easily if we add a check in `lib/augmentConfig.js#addPluginFunctions` to separate `string` and JS function types as `plugins` passed in the config and run the logic accordingly.

**Comment by ybiquitous** [comment]
@phoenisx Thanks for the proposal. As you commented, I also guess it's possible to implement this feature relatively easily.

I'd just like to confirm, what kind of use cases do you have? The following simple case seems to have not much benefit. 🤔 

```diff
-plugins: ["./plugin-path"],
+plugins: [require("./plugin-path")],
```

**Comment by ybiquitous** [comment]
@phoenisx Thank you for your explanation. I understand your background well. Also, it's so nice if you have a patch already.

Since I believe this addition would not break anything, I've labeled the issue as *ready to implement*.

If anyone has concerns, please feel free to comment.

**Comment by phoenisx** [comment]
Apologies for the delay, was a bit occupied.
I raised the PR with the required changes #6481 🙌🏽
